compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -809,7 +809,10 @@ object desugar {
             Nil
         }
       }
-      val classMods = if mods.is(Given) then mods &~ Given | Synthetic else mods
+      val classMods =
+        if mods.is(Given) then mods &~ Given | Synthetic | GivenClass
```

Comment:
Instead of dropping `Given` and adding a new `GivenClass`, what prevents us from keeping `Given`?